conjure-java-core/src/integrationInput/java/com/palantir/product/CollectionsTestObject.java

```diff
@@ -216,8 +217,12 @@ public Builder from(CollectionsTestObject other) {
         @JsonSetter(value = "items", nulls = Nulls.SKIP)
         public Builder items(@Nonnull Iterable<String> items) {
             checkNotBuilt();
-            this.items.clear();
-            ConjureCollections.addAll(this.items, Preconditions.checkNotNull(items, "items cannot be null"));
+            if (items instanceof Collection) {
+                this.items = new ArrayList<>((Collection) Preconditions.checkNotNull(items, "items cannot be null"));
+            } else {
+                this.items.clear();
+                ConjureCollections.addAll(this.items, Preconditions.checkNotNull(items, "items cannot be null"));
+            }
```

Comment:
this is an example of what could be much more concise with Guava's `Lists` `static <T> ArrayList<T> newArrayList(Iterable<T>)` or similar if we wanted to add to `ConjureCollections` 

```suggestion
            this.items = Lists.newArrayList(Preconditions.checkNotNull(items, "items cannot be null"));
```

Reply:
moved to `ConjureCollections`

Reply:
:+1: this allows hotspot to do what it's good at